### PR TITLE
Add a module for collecting iDRAC sensors

### PR DESCRIPTION
**Pull request by TuxPowered42**

### CODE REVIEW

**Comment by lamaral** [comment]
I think calling it lib_snmp.py might be better, so we make it explicit that this is not a standalone script.

**Comment by TuxPowered42** [comment]
> I think calling it lib_snmp.py might be better, so we make it explicit that this is not a standalone script.

Done.

**Review by Woellchen — COMMENTED** [review]
Let's make at least `flake8`/`pylint` happy with the two new files please

File: igcollect/idrac.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        # or amperageProbeTypeIsSystemWatts, the value returned for this attribute
+        # is the power usage that the probe is reading in Watts.
+        # When the value for amperageProbeType is other than amperageProbeTypeIsDiscrete,
+        # amperageProbeTypeIsPowerSupplyAmps, amperageProbeTypeIsPowerSupplyWatts,
+        # amperageProbeTypeIsSystemAmps or amperageProbeTypeIsSystemWatts,
+        # the value returned for this attribute is the amperage that the probe is
+        # reading in Milliamps.
+        # When the value for amperageProbeType is amperageProbeTypeIsDiscrete,
+        # a value is not returned for this attribute.
+        # amperageProbeTypeIsDiscrete(16),
```
[the remark this answers is not in the document]
Comment (TuxPowered42) [reply igcollect/idrac.py:51]: Done.

File: igcollect/idrac.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+def main():
+    timestamp = int(time())
+    args = parse_args()
+
+    try:
+        snmp = get_snmp_connection(args)
+    except error.PySnmpError as e:
+        print(e, file=sys.stderr)
+        return -1
```
[the remark this answers is not in the document]
Comment (TuxPowered42) [reply igcollect/idrac.py]: Removed handling of such exceptions.

File: igcollect/idrac.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    return parser.parse_args()
+
+
+def main():
+    timestamp = int(time())
+    args = parse_args()
+
+    try:
+        snmp = get_snmp_connection(args)
+    except error.PySnmpError as e:
```
[the remark this answers is not in the document]
Comment (TuxPowered42) [reply igcollect/idrac.py]: Removed handling of such exceptions.

File: igcollect/idrac.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+"""igcollect - idrac
+
+This script collects various parameters of iDRAC and a server it is in via SNMP.
+
+Copyright © 2023 InnoGames GmbH
+"""
+
+from argparse import ArgumentParser
+from time import time
+from pysnmp import error
```
[the remark this answers is not in the document]
Comment (TuxPowered42) [reply igcollect/idrac.py]: Fixed.

File: igcollect/idrac.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+        for probe_name, probe_reading in probe_data.items():
+            probe_name = probe_name.replace(' ', '_')
+            print(f'{prefix}.{probe_set}.{probe_name} {probe_reading} {timestamp}')
+
+
+def get_temperatures(snmp, config):
+    ret = {}
+    probe_units = get_snmp_table(snmp, config['probe_type'])
+    for probe_index, probe_data in get_probes_table(snmp, config).items():
+        if probe_units[probe_index] != 16:
```
[the remark this answers is not in the document]
Comment (TuxPowered42) [reply igcollect/idrac.py]: Fixed.

**Review by Woellchen — APPROVED** [review]

**Pull request merged**